ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * This file is intended to be renamed to metametrics.js once the conversion is complete.
+ * MetaMetrics is our own brand, and should remain aptly named regardless of the underlying
+ * metrics system. This file implements Segment analytics tracking.
+ */
+import React, { Component, createContext, useEffect, useCallback, useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import PropTypes from 'prop-types'
+import { useLocation, matchPath } from 'react-router-dom'
+import { captureException, captureMessage } from '@sentry/browser'
+
+import { pick } from 'lodash'
+import {
+  getCurrentNetworkId,
+  getAccountType,
+  getNumberOfAccounts,
+  getNumberOfTokens,
+} from '../selectors/selectors'
+import { getSendToken } from '../selectors/send'
+import {
+  txDataSelector,
+} from '../selectors/confirm-transaction'
+import { getEnvironmentType, getPlatform } from '../../../app/scripts/lib/util'
+import {
+  sendCountIsTrackable,
+  segment,
+  METAMETRICS_ANONYMOUS_ID,
+} from '../helpers/utils/metametrics.util'
+import { PATH_NAME_MAP } from '../helpers/constants/routes'
+
+export const MetaMetricsContext = createContext(() => {
+  captureException(
+    Error(`MetaMetrics context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
+  )
+})
+
+const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
+
+export function MetaMetricsProvider ({ children }) {
+  const txData = useSelector(txDataSelector) || {}
+  const network = useSelector(getCurrentNetworkId)
+  const activeCurrency = useSelector(getSendToken)?.symbol
+  const accountType = useSelector(getAccountType)
+  const confirmTransactionOrigin = txData.origin
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const numberOfTokens = useSelector(getNumberOfTokens)
+  const numberOfAccounts = useSelector(getNumberOfAccounts)
+  const location = useLocation()
+
+  const contextProperties = useMemo(() => ({
```

Comment:
they were being attached to all events for Matomo, but i doubt they will be in the tracking plan to be on every event. Do any of these stand out as things that make more sense as context instead of properties? I will remove the rest and put that responsibility on the downstream event calls. Didn't really like this API but didn't want to force more selector/hook usage elsewhere.

Reply:
Good question - I'm not sure.

`network` is fairly important to most metrics, as it lets us hone in on "real" transactions and exclude local and testnet stuff. So maybe that makes sense in context. I'm a little torn only because we intend to make network a per-dapp setting, but even still there might be a "current network" for any given UI state? Or at least most states. :thinking: We can cross that bridge when we come to it though. We can add a `N/A` value or something later if necessary.

`activeCurrency` is useful to know in general, but I don't know that we need it per-event? That seems like something that could go in `identify`. The same goes for `number_of_accounts` and `number_of_tokens`. `accountType` was meant to track how common hardware wallets were, and we could solve that with `identify` as well (e.g. `number_of_hardware_accounts` perhaps?).

`dapp_url` is critical for confirmations, but non-applicable anywhere else. I guess that could go in `context` too, and it'd just be blank for the non-applicable cases? I don't feel strongly as long as we track it somehow.

Reply:
`dapp_url` does show up in the mobile stuff as event properties, so probably not context. We could add it to the confirmation screens and include them in the events manually. That probably makes more sense.

When does `activeCurrency` change? 

the other variables I agree on identify calls as a means to track.

Already added network to context so it definitely doesn't need to be there. Cool.

Reply:
Oops, I was mixing up `activeCurrency` with `currentCurrency`.

`currentCurrency` is the user's preferred currency for display, and might be useful in `identify` along with the current locale.

`activeCurrency` appears to be the current selected currency when in the send flow specifically, which doesn't seem appropriate for context or identify. That seems like it should be explicitly passed in by the send flow events.

Reply:
resolved with https://github.com/MetaMask/metamask-extension/pull/9382/commits/34c92162e19d72a15eaaff438f4b94aae02ee9c7

Reply:
and i'll fix the lint issues I overlooked in the next commit ;)